examples/templates/ecs-container/main.tf

```diff
@@ -0,0 +1,242 @@
+}
+
+# provision auto-scaling group to host ECS task definitions
+resource "aws_autoscaling_group" "main" {
+  name                  = "coder-ecs-auto-scaling-group"
+  min_size              = 1
+  max_size              = 1
+  desired_capacity      = 1
+  availability_zones    = ["us-east-1a"]
+  protect_from_scale_in = true
+
+  launch_template {
+    id      = aws_launch_template.coder-oss-ubuntu.id
+    version = "$Latest"
+  }
+
+  tag {
+    key                 = "AmazonECSManaged"
+    value               = true
+    propagate_at_launch = true
+  }
+}
+
+# create AWS ECS cluster
+resource "aws_ecs_cluster" "main" {
+  name = "coder-oss-ecs"
+
+  setting {
+    name  = "containerInsights"
+    value = "enabled"
+  }
+}
+
+# create capacity provider & tie in autoscaling group
+resource "aws_ecs_capacity_provider" "main" {
+  name = "coder-oss-capacity-provider"
+
+  auto_scaling_group_provider {
+    auto_scaling_group_arn         = aws_autoscaling_group.main.arn
+    managed_termination_protection = "ENABLED"
+
+    managed_scaling {
+      maximum_scaling_step_size = 1000
+      minimum_scaling_step_size = 1
+      status                    = "ENABLED"
+      target_capacity           = 10
+    }
+  }
+}
+
+# attach capacity provider to cluster
+resource "aws_ecs_cluster_capacity_providers" "main" {
+  cluster_name = aws_ecs_cluster.main.name
+
+  capacity_providers = [aws_ecs_capacity_provider.main.name]
+}
+
+# coder workspace, created as an ECS task definition
+resource "aws_ecs_task_definition" "workspace" {
+  family = "coder"
+
+  requires_compatibilities = ["EC2"]
+  cpu                      = 1024
+  memory                   = 2048
+  container_definitions = jsonencode([
+    {
+      name      = "coder-workspace-1"
+      image     = "codercom/enterprise-base:ubuntu"
+      cpu       = 1024
+      memory    = 2048
+      essential = true
+      user      = "coder"
+      command   = ["sh", "-c", coder_agent.coder.init_script]
+      environment = [
+        {
+          "name"  = "CODER_AGENT_TOKEN"
+          "value" = coder_agent.coder.token
+        }
+      ]
+      mountPoints = [
+        {
+          # the name of the volume to mount
+          sourceVolume = "home-dir"
+          # path on the container to mount the volume at
+          containerPath = "/home/coder"
+        }
+      ]
+      portMappings = [
+        {
+          containerPort = 80
+          hostPort      = 80
+        }
+      ]
+    }
+  ])
+
+  # workspace persistent volume definition
+  volume {
+    name = "home-dir"
+
+    docker_volume_configuration {
+      # "shared" ensures that the disk is persisted upon workspace restart
+      scope         = "shared"
+      autoprovision = true
+      driver        = "local"
+    }
+  }
+}
+
+resource "aws_ecs_service" "workspace" {
+  name            = "workspace"
+  cluster         = aws_ecs_cluster.main.id
+  task_definition = aws_ecs_task_definition.workspace.arn
+  desired_count   = 1
```

Comment:
```suggestion
  # Scale the service to zero when the workspace is stopped
  desired_count = data.coder_workspace.me.start_count
```

This will scale the service to 0 when the workspace is stopped and to 1 when started. 

ex. stopped state:
![esc-scale-service](https://user-images.githubusercontent.com/22407953/188751027-c15b511c-3589-4e9c-b9ba-6fb213871acf.png)